function-app/adb-to-purview/src/Function.Domain/Providers/SynapseClientProvider.cs

```diff
@@ -249,7 +250,57 @@ public async Task<string> GetSparkNotebookSource(string synapseWorkspaceName, st
 
             }
 
+        }
+
+        public async Task<string> GetSynapseStorageLocation(string synapseWorkspaceName, string databaseTableName)
+        {
+            if (_cache.TryGetValue<string>(databaseTableName, out string? cachedSynapseStorageLocation))
+            {
+                return cachedSynapseStorageLocation;
+            }
+
+            if (isTokenExpired(_bearerToken))
+            {
+                await GetBearerTokenAsync();
+
+                if (_bearerToken is null)
+                {
+                    _log.LogError("SynapseClient-GetSynapseJobAsync: unable to get bearer token");
+                    return null;
+                }
+            }
+
+            var values = databaseTableName.Split('/');
```

Comment:
It seems that since we need to split here, why not just have this provider accept database name and table name as distinct parameters. any concatenation of these values for caching purpose can be internal logic to this class.

Reply:
Done